appsmithorg/appsmith
# feat: gsheet disable new query when no files selected

## Description

This PR adds:
- When gsheet datasource is created with "Selected Sheets" Modality, and if user fails to select any files from file picker
    - we should disable new query creation
    - we should show error message banner on datasource review page

**How to test:**
- Create google sheet datasource, break the authorisation flow, check review page, error message like "Data source is not authorized, please authorize to continue." would be shown.
- Edit datasource, complete the authorisation but do not pick any files from file picker, error message like "Datasource does not have access to any files, please authorize google sheets to use this data source" would be shown


Fixes #20290

Media
> A video or a GIF is preferred. when using Loom, don’t embed because it looks like it’s a GIF. instead, just link to the video


## Type of change

> Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Manual

### Test Plan
> Add Testsmith test cases links that relate to this PR

### Issues raised during DP testing
> Link issues raised during DP testing for better visiblity and tracking (copy link from comments dropped on this PR)


## Checklist:
### Dev activity


### QA activity:
